src/tir/schedule/primitive/compute_at.cc

```diff
@@ -0,0 +1,584 @@
+    // Check and update the cached `var_dom`
+    runtime::StorageScope scope =
+        relax_storage_scope ? runtime::StorageScope::Create(buffer.scope()) : global_scope;
+    runtime::StorageRank rank = scope.rank;
+    if (rank != previous_rank || !var_dom.defined()) {
+      previous_rank = rank;
+      var_dom = AsIntSet(LoopDomainOfSRefTreePath(
+          /*low_inclusive=*/relax_path_low_inclusive,
+          /*high_exclusive=*/relax_path_high_exclusive,
+          /*extra_relax_scope=*/scope));
+    }
+    // Relax the region
+    Array<arith::IntSet> relaxed = arith::EvalSet(Substitute(region, binding), var_dom.value());
+    relaxed_regions.push_back({relaxed.begin(), relaxed.end()});
+  }
+}
+
+/*!
+ * \brief Calculate the iteration domain of a provided integer set to fully cover the required
+ * domain
+ * \param provided The provided integer set to cover the required domain
+ * \param required The required domain to be covered
+ * \param iter_doms The result iteration domains to be updated
+ * \param analyzer The arithmetic analyzer
+ */
+void UpdateBlockVarDomain(const arith::IntSet& provided, const arith::IntSet& required,
+                          std::unordered_map<const VarNode*, std::vector<arith::IntSet>>* iter_doms,
+                          arith::Analyzer* analyzer) {
+  PrimExpr provided_min = analyzer->Simplify(provided.min());
+  PrimExpr provided_extent = analyzer->Simplify(provided.max() - provided_min + 1);
+  PrimExpr required_min = analyzer->Simplify(required.min());
+  PrimExpr required_extent = analyzer->Simplify(required.max() - required_min + 1);
+  PrimExpr dom_min{nullptr}, dom_extent{nullptr};
+  Var dom_var{ObjectPtr<VarNode>{nullptr}};
+  arith::PVar<Var> p_v;
+  arith::PVar<PrimExpr> p_e;
+  if ((p_v * p_e).Match(provided_min) || (p_e * p_v).Match(provided_min)) {
+    PrimExpr e = p_e.Eval();
+    dom_var = p_v.Eval();
+    dom_min = floordiv(required_min, e);
+    dom_extent = analyzer->Simplify((required_extent + e - 1) / e);
+  } else if (analyzer->CanProveEqual(provided_extent, 1) && p_v.Match(provided_min)) {
+    dom_var = p_v.Eval();
+    dom_min = required_min;
+    dom_extent = required_extent;
+  } else {
+    ICHECK(false) << "ValueError: BufferRegion pattern match failed";
+  }
+  auto it = iter_doms->find(dom_var.get());
+  if (it != iter_doms->end()) {
+    std::vector<arith::IntSet>& doms = it->second;
+    doms.push_back(arith::IntSet::FromMinExtent(dom_min, dom_extent));
+  } else {
+    ICHECK(analyzer->CanProveEqual(provided_min, required_min));
+    ICHECK(analyzer->CanProveEqual(provided_extent, required_extent));
+  }
+}
+
+/*!
+ * \brief Calculate the domain of block vars to cover the required region
+ * \param iter_vars The list of block vars to cover the required region
+ * \param provided_regions The region provided by one iteration instance of the block vars
+ * \param required_regions The region required to be covered
+ * \param analyzer The arithmetic analyzer
```

Comment:
The `analyzer` was previously updated in `CheckAndBindLoopDomain(..)` via `analyzer->Bind(...)`, so I wonder whether it will be good to tell in the document that what kind of information `analyzer` has.

Reply:
To be clear, this method doesn't mutate `analyzer`, and not binding analyzer won't cause any problem internally in this method - the method doesn't depend on any property of the analyzer.